README.md

```diff
@@ -1 +1,60 @@
-# narayana-proposals
\ No newline at end of file
+# Narayana Proposals
+
+## Introduction
+
+This workflow provides a guideline to formally define the stages to propose, design, implement, and merge new Requests for Enhancement (RFEs) in the Narayana project. The process defined here can be followed by any contributor who would like to reach pre-agreement for the design they would like to use for the implementation of RFEs in the Narayana project. On the other hand, the employment of this workflow is _highly recommended_ when addressing larger scoped or complex features, and when there are disagreements among the community members of Narayana. If contributors have any uncertainties about whether this workflow is appropriate for their RFE, we strongly recommend that they reach out to the team (through the official Social Channels) to determine whether their investigation area would benefit from sharing design documents before proceeding with the development of their solution. Failing to do so might result in the realisation that the design is incompatible with Narayana's architecture.
+
+The ultimate goal of this workflow is the optimisation of the proposal management of the Narayana project, in the effort of achieving a smoother transition of new RFEs from conceptualisation to realisation while fostering collaboration and knowledge sharing in our community.
+
+## 0. Secure the approval
+
+One of the fundamental improvements introduced by this workflow is the implementation of a process to secure the approval of the Narayana team (which consists of the [Contributors](https://www.narayana.io/community/index.html) in “Our team”, who are responsible for maintaining the Narayana code base). To progress a proposal, the whole Narayana team is required to approve it within a defined time window, which is set to 14 days[^1]. Optional members - specifically [Members](https://github.com/orgs/jbosstm/people?query=role%3Amember) and [Outside Contributors](https://github.com/orgs/jbosstm/outside-collaborators) - have the flexibility to participate at their discretion and their opinion is equal in importance to the [Narayana team](https://www.narayana.io/community/index.html). As a consequence, every contributor from the aforementioned groups (hereafter “reviewers”) can veto the approval of a proposal within the defined time window.
```

Comment:
The proposal still contains too many holes and is too complex and prescriptive and the review is likely to drag on. Since I appear to be in a minority I shall abstain and I'll withdraw my initial review but only until you remove the requirement for the whole of the core team to review it within 3 days.

Reply:
I will remove the 3 days, no problem. Thanks Michael :-)

Reply:
If you need it to be 3 days then just say "Reviews by team members must be submitted within 3 days but it is not mandatory for all team members to provide a review", but factor in the need for engineers to take PTO .

Oh, and I forgot to add that not being able to "address the implementation of the RFE" when the ultimate PR for the feature is raised is not practical - it is invitation for the author to ignore reviewers' comments. So I need that to be resolved before withdrawing my review.

Reply:
> If you need it to be 3 days then just say "Reviews by team members must be submitted within 3 days but it is not mandatory for all team members to provide a review", but factor in the need for engineers to take PTO .

Yep, that could be specified in the text.

> Oh, and I forgot to add that not being able to "address the implementation of the RFE" when the ultimate PR for the feature is raised is not practical - it is invitation for the author to ignore reviewers' comments. So I need that to be resolved before withdrawing my review.

This should not be the case. Even when the technical document and the implementation document are approved, reviewers must carry out a _code_ review when the PR is raised. This means that reviewers can and should point out all _code_ issues related to the actual implementation, for example (but not limited to) bugs, anti-patterns, code smells, CVEs, formatting, spelling mistakes. What reviewers of the PR should _not_ discuss during the code review is the design already approved with the technical and implementation documents. In other words, the PR can be debugged through multiple rounds of _code_ reviews but the overall _design_ and _implementation choices_ of the PR should not be discussed as approvals should have been already granted at this stage.